File: mns-reactor/src/test/java/cz/fb/manaus/reactor/betting/BetCollectorTest.java
package cz.fb.manaus.reactor.betting;

import cz.fb.manaus.core.model.Bet;
import cz.fb.manaus.core.model.BetAction;
import cz.fb.manaus.core.model.Price;
import cz.fb.manaus.core.model.Side;
import org.junit.Test;

import static junit.framework.TestCase.assertFalse;
import static org.junit.Assert.assertTrue;

public class BetCollectorTest {

    public static final String MARKET_ID = "44";
    public static final int SELECTION_ID = 111;

    @Test
    public void testFindBet() {
        var updateBet = new Bet("777", MARKET_ID, SELECTION_ID, new Price(5d, 5d, Side.LAY), null, 0d);
        var collector = new BetCollector();
        collector.updateBet(new BetCommand(updateBet, new BetAction()));
        checkCollector(collector);
        collector = new BetCollector();
        var placeBet = new Bet(null, MARKET_ID, SELECTION_ID, new Price(5d, 5d, Side.LAY), null, 0d);
        collector.placeBet(new BetCommand(placeBet, new BetAction()));
        checkCollector(collector);
    }

    private void checkCollector(BetCollector collector) {
        assertTrue(collector.findBet(MARKET_ID, SELECTION_ID, Side.LAY).isPresent());
        assertFalse(collector.findBet(MARKET_ID, SELECTION_ID, Side.BACK).isPresent());
        assertFalse(collector.findBet(MARKET_ID + 1, SELECTION_ID, Side.LAY).isPresent());
        assertFalse(collector.findBet(MARKET_ID, SELECTION_ID + 1, Side.LAY).isPresent());
    }

    @Test
    public void testEmpty() {
        var collector = new BetCollector();
        assertTrue(collector.isEmpty());
    }
}

File: mns-reactor/src/test/java/cz/fb/manaus/reactor/betting/BetCollectorTest.kt
package cz.fb.manaus.reactor.betting

import cz.fb.manaus.core.model.Bet
import cz.fb.manaus.core.model.Price
import cz.fb.manaus.core.model.Side
import cz.fb.manaus.core.model.betAction
import org.junit.Test
import kotlin.test.assertNotNull
import kotlin.test.assertNull
import kotlin.test.assertTrue

class BetCollectorTest {
    private val marketId = "44"
    private val selectionId: Long = 111

    @Test
    fun `find bet`() {
        val updateBet = Bet(betId = "777", marketId = marketId, selectionId = selectionId,
                requestedPrice = Price(5.0, 5.0, Side.LAY))
        var collector = BetCollector()
        collector.updateBet(BetCommand(updateBet, betAction))
        checkCollector(collector)
        collector = BetCollector()
        val placeBet = Bet(betId = null, marketId = marketId, selectionId = selectionId,
                requestedPrice = Price(5.0, 5.0, Side.LAY))
        collector.placeBet(BetCommand(placeBet, betAction))
        checkCollector(collector)
    }

    private fun checkCollector(collector: BetCollector) {
        assertNotNull(collector.findBet(marketId, selectionId, Side.LAY))
        assertNull(collector.findBet(marketId, selectionId, Side.BACK))
        assertNull(collector.findBet(marketId + 1, selectionId, Side.LAY))
        assertNull(collector.findBet(marketId, selectionId + 1, Side.LAY))
    }

    @Test
    fun `empty collector`() {
        val collector = BetCollector()
        assertTrue(collector.isEmpty)
    }
}
